packages/framework-core/src/decorators/projects.ts

```diff
@@ -105,7 +105,7 @@ type ProjectionMethod<
 > = TJoinKeyType extends ReadModelJoinKeyFunction<TEntity, TReadModel>
   ? ProjectionMethodWithEntityConditionalReadModelIdAndReadModel<TEntity, TReadModel>
   : TJoinKeyType extends keyof TEntity
-  ? PropertyType<TEntity, TJoinKeyType> extends Array<UUID>
+  ? NonNullable<PropertyType<TEntity, TJoinKeyType>> extends Array<any>
```

Comment:
I think that the `any` modifier should be used as the last resource in TypeScript. Have you tried these approaches before choosing `any` as the solution?

(1) Maybe is suitable to create a union type for the allowed types for this type, something like:
```typescript
type AllowedPropertyType = UUID | string //we should find a better name for this

...
? NonNullable<PropertyType<TEntity, TJoinKeyType>> extends Array<AllowedPropertyType>
```

(2) another option is to try the `unknown` type instead of `any`

If both options above don't work, let's go for `any`

Reply:
@davidverdu look at commit 51dfedc. I added a `UUIDLike` type following your suggestion. :wink: :+1: